runtime/lib/ttmetal/runtime.cpp

```diff
@@ -123,50 +130,138 @@ void closeDevice(Device device) {
   }
 }
 
+static std::pair<std::shared_ptr<::tt::tt_metal::Buffer>,
+                 std::shared_ptr<::tt::tt_metal::Event>>
+prepareInput(::tt::tt_metal::Device *device, MetalTensor const &metalTensor,
+             void *data, ::tt::target::TensorRef const *tensorRef) {
+  if (TensorDesc const *hostTensorDesc = std::get_if<TensorDesc>(&metalTensor);
```

Comment:
can we use `std::holds_alternative<TensorDesc>(metalTensor)` instead (more readable imo)?

Reply:
Done!